tools/links-check/index.ts

```diff
@@ -9,37 +13,57 @@ main().catch((err: unknown) => {
 });
 
 async function main() {
-  const args = process.argv.slice(2);
+  program
+    .showHelpAfterError()
+    .argument("<paths...>", "Paths of files to be scanned. Supports glob patterns.")
+    .option("--ignore-file <path>", "Path to the ignore file.")
+    .action(async (paths, options) => {
+      let files = await fastGlob(paths);
+      const commonPath = getCommonPath(files);
 
-  if (args.length === 0) {
-    throw new Error("Provide a list of files to scan.");
-  }
+      if (options.ignoreFile) {
+        let ignorePatterns: string[];
 
-  const files = await fastGlob(args);
-  const commonPath = getCommonPath(files);
+        try {
+          const ignoreFileContent = await fs.promises.readFile(path.resolve(options.ignoreFile), "utf-8");
+          ignorePatterns = ignoreFileContent.split("\n").filter(Boolean);
+          console.info(`Using ignore file: ${options.ignoreFile}`);
+        } catch (err) {
+          console.error(`Failed to read ignore file: ${err}`);
+          process.exit(1);
+        }
 
-  console.info();
+        if (ignorePatterns.length) {
+          const ignoreManager = ignore().add(ignorePatterns);
+          files = files.filter((file) => !ignoreManager.ignores(path.relative(commonPath, file)));
+        }
+      }
 
-  const metadata = await fetchMetadata();
+      console.info();
 
-  const label = `scanning ${files.length}`;
-  console.time(label);
-  let report: Report | null = null;
-  try {
-    report = await scan(files, metadata, commonPath);
-  } finally {
-    console.timeEnd(label);
-  }
-  console.info();
-  console.info();
+      const metadata = await fetchMetadata();
 
-  if (!report) {
-    return;
-  }
+      const label = `scanning ${files.length}`;
```

Comment:
```suggestion
      const label = `scanning ${files.length} (${ignored} ignored)`;
```

Can we additionally display number of ignored files in case `--ignore-file` flag is present? I think it will be easier to understand what's happening when you end up with `scanning 0`.

Reply:
done